src/ray/gcs/gcs_server/gcs_kv_manager.cc

```diff
@@ -14,10 +14,37 @@
 
 #include "ray/gcs/gcs_server/gcs_kv_manager.h"
 
+#include <string_view>
+
 #include "absl/strings/match.h"
+#include "absl/strings/str_split.h"
 
 namespace ray {
 namespace gcs {
+namespace {
+constexpr std::string_view kNamespacePrefix = "@namespace_";
+constexpr std::string_view kNamespaceSep = ":";
+
+std::string MakeKey(const std::string &ns, const std::string &key) {
+  if (ns.empty()) {
+    return key;
+  }
+  return absl::StrCat(kNamespacePrefix, ns, ":", key);
+}
+
+std::string ExtractKey(const std::string &key) {
+  if (absl::StartsWith(key, kNamespacePrefix)) {
+    std::vector<std::string> parts =
+        absl::StrSplit(key, absl::MaxSplits(kNamespaceSep, 1));
+    if (parts.size() != 2) {
+      RAY_LOG(ERROR) << "Invalid key: " << key;
+      return "";
```

Comment:
This shouldn't happen with Max splits set right? Check fail instead?

Reply:
I think it's possible, for example, the user use:
`@namespace_` as the key with ns empty.

Maybe we should forbid this key from putting into kv.

Reply:
But absl::MaxSplits will limit the splits to 1...

Reply:
I don't think so. MaxSplits means split to at most limits + 1. (https://github.com/abseil/abseil-cpp/blob/master/absl/strings/str_split.h#L281-L284)

I tested and for this `a:b:c` and max split with `:,1`, it'll be `[a, b:c]`

for `a` and max split with `:,1`, it'll be `[a]`

Reply:
I also thought about check in cpp, but it'll bright issues with backward compatible since the old version of ray will send `@namespace_something:key` to gcs.

But I think it's not an issue since anyway for ray client, we don't support namespace right now, so namespace should always be "" for them.

For autoscaler, we also don't use namespace right now.

Reply:
So should we check if it's >2 and warn if it's 1?

Reply:
max split guarantees it's <= 2. This is to check the key error internally.
I added key validation in the code and guarantee it's always 2.